File: 117_looping_advanced/lesson/pola_5.cpp
#include <iostream>

using namespace std;
int factorial(int n);
int c(int n, int k);

int main() {
  int i, j, jml;
  int n, k, coef;
  cout << "Masukkan jumlah: " << endl;
  cin >> jml;
  
  for (i = 0; i <= jml; i++) {
    n = 1;
    // for (j = 1; j < (jml - i); j++) {
    //   // cout << "   ";
    //   cout << " * ";
    // }
    // for (k = 0; k <= i; k++) {
    //   // cout << "      " << "*";
    //   cout << " " << n << " ";
    //   n = n * (i - k) / (k + 1); // rumus pascal triangle
    // }
    coef = c(jml, i);
    cout << " " << coef << " ";
    cout << endl << endl;
  }
}

/*
  Factorial function
*/

int factorial(int n) {
  if (n == 0) {
    return 1;
  } else {
    return n * factorial(n - 1);
  }
}

int c(int n, int k) {
  return factorial(n) / (factorial(k) * factorial(n - k));
}

File: 117_looping_advanced/lesson/pola_5.pas
program pola_5;

var
  i, j: integer; // looping
  jml: integer; // total
  n, k: integer; // baris dan kolom

function fact(n: integer): longint;
{ Fungsi rekursif, mencari faktorial }
begin
  if (n = 0) then begin
    fact := 1;
  end else begin
    fact := n * fact(n - 1);
  end;
end;

function c(n, k: integer): longint;
{ n = baris, k = kolom }
begin
  c := fact(n) div (fact(k) * fact(n - k)); // rumus segitiga (pascal: nama orang) 
end;

begin
  write('masukkan jumlah: '); readln(jml);
  n := -1;
  for i := jml downto 1 do begin
    n := n + 1;
    k := -1; // reset kolom
    for j := 1 to jml do begin
      if (j >= i) then begin
        k := k + 1;
        write(c(n, k));
      end; // only show number when triggered
      write(' ');
    end; // end j
    writeln;
  end; // end i
end.